File: Placement Sheet/Arrays/Service Based/Make_Unique_Array.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class Make_Unique_Array {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        ArrayList<Integer> arr = new ArrayList<>();
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the list: ", i + 1);
            arr.add(input.nextInt());
        }
        input.close();
        System.out.println(minElementsToRemove(arr));
    }

    private static int minElementsToRemove(ArrayList<Integer> arr) {
        if (arr.isEmpty()) {
            return 0;
        }
        Set<Integer> set = new HashSet<>();
        int count = 0;
        for (int i = 0; i < arr.size(); i++) {
            int currElm = arr.get(i);
            if (set.contains(currElm)) {
                count++;
                continue;
            }
            set.add(currElm);
        }
        return count;
    }
}

File: Placement Sheet/Arrays/Initial Startups/Python/Make_Unique_Array.py
def minIncrementForUnique(arr):
    try:
        arr.sort()
        count = 0
        for i in range(1, len(arr)):
            if arr[i] <= arr[i - 1]:
                count += arr[i - 1] - arr[i] + 1
                arr[i] = arr[i - 1] + 1
        
        return count
    
    except Exception as e:
        print(f"Error: {e}")

size = int(input("Enter the size of the array: "))
arr = []

for i in range(0, size):
    data = int(input("Enter the data: "))
    arr.append(data)

print(minIncrementForUnique(arr=arr))